src/devices/speechSynthesizer_nwc_yarp/SpeechSynthesizer_nwc_yarp.cpp

```diff
@@ -0,0 +1,194 @@
+/*
+ * SPDX-FileCopyrightText: 2023-2023 Istituto Italiano di Tecnologia (IIT)
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include "SpeechSynthesizer_nwc_yarp.h"
+
+#include <yarp/os/LogComponent.h>
+#include <yarp/os/LogStream.h>
+#include <yarp/os/Os.h>
+
+namespace {
+YARP_LOG_COMPONENT(SPEECHSYNTH_NWC, "yarp.devices.speechSynthesizer_nwc_yarp")
+}
+
+SpeechSynthesizer_nwc_yarp::~SpeechSynthesizer_nwc_yarp()
+{
+    closeMain();
+}
+
+bool SpeechSynthesizer_nwc_yarp::close()
+{
+    return closeMain();
+}
+
+bool SpeechSynthesizer_nwc_yarp::open(yarp::os::Searchable& config)
+{
+    m_local_name.clear();
+    m_thrift_server_name.clear();
+
+    if(!config.check("local"))
+    {
+        yCError(SPEECHSYNTH_NWC) << "No local name specified";
+
+        return false;
+    }
+    m_local_name = config.find("local").asString();
+
+    if(!config.check("remote"))
+    {
+        yCError(SPEECHSYNTH_NWC) << "No remote name specified";
+
+        return false;
+    }
+    m_thrift_server_name = config.find("remote").asString();
+
+    if(config.check("carrier"))
```

Comment:
not sure about this, but I think that the carrier option is typically used for not-rpc (i.e. streaming) communication. does it make sense to do an udp rpc? probably it's risky..